assign_nodes.py

```diff
@@ -24,40 +26,91 @@ def unique_nodes(examples):
 
 def find_node(node, kg, ontology = ""):
 	nodes = kg.labels_all
+	### All caps input is probably a gene or protein. Either search in a case sensitive manner or assign to specific ontology. 
+
 
 	if ontology == "":
-		results = nodes[nodes["label"].str.contains(node,flags=re.IGNORECASE, na = False)|nodes["description/definition"].str.contains(node,flags=re.IGNORECASE, na = False)][["integer_id","label","entity_uri", "description/definition"]]
+		if node.isupper(): #likely a gene or protein
+			results = nodes[(nodes["label"].str.contains(node,flags=re.IGNORECASE, na = False)|nodes["description/definition"].str.contains(node,flags=re.IGNORECASE, na = False)) & nodes["entity_uri"].str.contains("gene|PR|GO",flags=re.IGNORECASE, na = False) ][["integer_id","label", "description/definition"]]
+		else:
+			results = nodes[nodes["label"].str.contains(node,flags=re.IGNORECASE, na = False)|nodes["description/definition"].str.contains(node,flags=re.IGNORECASE, na = False)][["integer_id","label","entity_uri", "description/definition"]]
 	else:
 		results = nodes[(nodes["label"].str.contains(node,flags=re.IGNORECASE, na = False)|nodes["description/definition"].str.contains(node,flags=re.IGNORECASE, na = False)) & nodes["entity_uri"].str.contains(ontology,flags=re.IGNORECASE, na = False) ][["integer_id","label", "description/definition"]]
 	return(results)
 
 # Create a list of nodes for input
-def search_nodes(nodes, kg):
-	user_nodes = []
+
+# Could potentially find several features for a single input example. Need a way to be able to select multiple feaures for a search. 
+# Need a way to go back through search terms. 
+
+def search_nodes(nodes, kg, examples):
+	examples["source_label"] = ""
+	examples["target_label"] = ""
 	for node in nodes:
 		print("User Search Node: ", node)
-		found_nodes = find_node(node,kg)
-		
+		found_nodes = find_node(node,kg)		
 		nrow = found_nodes.shape[0]
+		print("Found", nrow, "features in KG")
+		user_input = ""
+		bad_input = True
 		if nrow < 20:
-			print(found_nodes.iloc[0:20,])
-			user_input = input("Input node integer_id: ")
-		else:
-			n = range(0, nrow, 20)
-			user_input = ""
-			i = 0
-			while(type(user_input) != int):
-				print(found_nodes.iloc[n[i]:n[i+1],])
-				user_input = input("Input node integer_id or hit enter for the next 20 features: ")
+			while(bad_input):
+				print(found_nodes.iloc[0:nrow,])
```

Comment:
perhaps don't print the index # so it is easy for the user to distinguish which number to enter (integer_id)